design/design-doc-stacks.md

```diff
@@ -301,24 +386,25 @@ Possible solutions for package processing are listed below, along with their pro
   * PRO: Package images are significantly smaller since they will only contain yaml files and icons, no binaries or libraries.
+  * PRO: The processing/unpacking logic can easily move to its own image that can be used as a base layer in the future (or a CLI tool), so this approach is not very divergent.
   * CON: The SM needs to understand package content and layout.
 * CLI tool
   * The unpacking logic could be built into a CLI tool that can be run over a package image.
   * PRO: This is the most versatile option as it can be used in contexts outside of a Crossplane cluster.
   * CON: Doesn't integrate very cleanly into the Crossplane mainline scenario with the SM.
 
-**Recommendation:** It is recommended to start with the "Job initContainer and shared volume" approach that copies the package contents from the package initContainer to a shared volume and has a Crossplane container main entry point perform the processing logic over the shared volume contents, sending out the artifacts to `stdout`.
-Stack packages should only have a few KB of yaml files, so it's a significant concern to blow that image size up to many MB right now.
-The processing/unpacking logic can easily move to its own image that can be used as a base layer in the future (or a CLI tool), so this approach is not very divergent.
-It is a good place to start and iterate over as we learn more without investing much effort that cannot be reused.
+Each of these designs offered a good place to start.  Through iteration over time we will learn more, hopefully without investing much effort that cannot be reused.
 
 ## Questions and Open Issues
 
 * Dependency resolution design: [#434](https://github.com/crossplaneio/crossplane/issues/434)
-* Updating/Upgrading Extension: [#435](https://github.com/crossplaneio/crossplane/issues/435)
+* Updating/Upgrading Extensions: [#435](https://github.com/crossplaneio/crossplane/issues/435)
 * Document how to build and publish an extension [#503](https://github.com/crossplaneio/crossplane/issues/503)
 * Support installation of extensions from private registries [#505](https://github.com/crossplaneio/crossplane/issues/505)
 * Standardize extension package format and layout [#507](https://github.com/crossplaneio/crossplane/issues/507)
 * Figure out model for crossplane core vs extensions [#531](https://github.com/crossplaneio/crossplane/issues/531)
 * Single extension should be able to install multiple controllers [#532](https://github.com/crossplaneio/crossplane/issues/532)
 * Prototype alternate extension implementations [#548](https://github.com/crossplaneio/crossplane/issues/548)
+* Is there a benefit to `kind.version.` prefixed `crd.yaml` filenames
+* What categories are valid? Is there a well-defined Category tree? Are arbitrary categories invalid or ignored?
+* Should links be predefined (`website`, `source`) or freeform `links:[{description:"Website",url:"..."}, ...]`?
```

Comment:
Oh, this seems to answer my earlier question about links o:

Reply:
I'm open to more links in the spec, but if this goes in a ui, description isn't really a key, and specific keys have a meaning to be surfaced in a UI or maybe validated, and it just feels less elegant to say you might need to give a description that's more like a known key to show up in the UI, although this would still work.

I think ad-hoc any category could be valid, but someone might choose to only surface a set of preferred categories in browse and choose to ignore the rest that don't conform to the predefined list.

Reply:
+1 in favor of standard fields (i.e. "Supply values here and any UI can guarantee that they will be present or blank") with the option to add ad-hoc field as well (i.e. "I am a UI developer and I will parse out this field that is provided in an ad-hoc manner if it is present and show it")